homeassistant/components/plex/config_flow.py

```diff
@@ -0,0 +1,215 @@
+                    CONF_TOKEN, default=self.current_login.get(CONF_TOKEN, "")
+                ): str,
+                vol.Optional("manual_setup"): bool,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_manual_setup(self, user_input=None):
+        """Begin manual configuration."""
+        if user_input is None:
+            data_schema = vol.Schema(
+                {
+                    vol.Required(
+                        CONF_HOST, default=self.discovery_info.get(CONF_HOST)
+                    ): str,
+                    vol.Required(
+                        CONF_PORT,
+                        default=int(self.discovery_info.get(CONF_PORT, DEFAULT_PORT)),
+                    ): int,
+                    vol.Optional(CONF_SSL, default=DEFAULT_SSL): bool,
+                    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): bool,
+                    vol.Optional(CONF_TOKEN): str,
+                }
+            )
+            return self.async_show_form(step_id="manual_setup", data_schema=data_schema)
+
+        host = user_input.pop(CONF_HOST)
+        port = user_input.pop(CONF_PORT)
+        prefix = "https" if user_input.get(CONF_SSL) else "http"
+        user_input[CONF_URL] = f"{prefix}://{host}:{port}"
+        return await self.async_step_user(user_input=user_input)
+
+    async def async_step_select_server(self, user_input=None):
+        """Use selected Plex server."""
+        if user_input is None:
+            return await self.async_step_user()
+
+        config = self.current_login
+        config[CONF_SERVER] = user_input.get(CONF_SERVER)
+        return await self.async_step_user(user_input=config)
+
+    async def async_step_discovery(self, discovery_info):
+        """Set default host and port from discovery."""
+        if self._async_in_progress():
+            return self.async_abort(reason="already_in_progress")
+
+        self.discovery_info = discovery_info
+        return await self.async_step_user()
+
+    async def async_step_import_plex_conf(self, import_config):
+        """Import from Plex media_player file config.
+
+        Legacy.
+        """
+        if self._async_in_progress():
+            return self.async_abort(reason="already_in_progress")
+
+        host_and_port, host_config = import_config.popitem()
+        prefix = "https" if host_config[CONF_SSL] else "http"
+        url = f"{prefix}://{host_and_port}"
+
+        config = {
+            CONF_URL: url,
+            CONF_TOKEN: host_config[CONF_TOKEN],
+            CONF_VERIFY_SSL: host_config["verify"],
+        }
+
+        _LOGGER.info("Imported configuration from legacy config file")
+        return await self.async_step_user(user_input=config)
+
+    async def async_step_import(self, import_config):
+        """Import from Plex configuration."""
+        if self._async_in_progress():
```

Comment:
Shouldn't this entry point be priority over discovery step?

Reply:
Yes, but setup/import will always fire before discovery.

Reply:
So then we don't need this check?

Reply:
Perhaps a user is creating a flow slowly? 😁

Reply:
But the import step will happen during home assistant start. It's unlikely that the user is able to start a flow before this step completes. I'm not sure it's possible.